### PR TITLE
[Tests][Enhancement] Fix pooling command printing

### LINKED ISSUE #1647: [driver][pool] #1313 is missing fixes in Pooling_logging_cmd().
#1313 is missing some changes in `Pooling_logging_cmd()`. Example of symptoms: https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1642#issuecomment-1192114173.

@junliume @DrizztDoUrden This is a bug of normal priority (formally) because it affects debugging only, but...

### PR DESCRIPTION
**Pull request by DrizztDoUrden**
Command is now printed with args in a driver-accepted format, rather than old non supported one.
Fixes #1647.

### CODE REVIEW

**Comment by DrizztDoUrden** [comment]
@atamazov I can't set you as a reviewer, but here we are.

**Comment by junliume** [comment]
@DrizztDoUrden  I built your branch and tried to test locally:
```
/root/MIOpen/build# ./bin/MIOpenDriver pool -M 0 -n 2 -c 2 -H 1 -W 5 -y 1 -x 1 -p 0 -q 0 -v 1 -u 1 -m avg -F 1 -t 1
MIOpenDriver pool -M 0 -n 2 -c 2 -H 1 -W 5 -y 1 -x 1 -p 0 -q 0 -v 1 -u 1 -m avg -F 1 -t 1
Input Flag: n Not Found !
```
anything I might have missed here?

File: src/pooling_api.cpp [hunk 1]
```diff
@@ -41,30 +41,37 @@ inline void Pooling_logging_cmd(const miopenPoolingDescriptor_t poolDesc,
...
-        {
-            ss << "poolfp16";
-        }
-        else
-        {
-            ss << "pool";
-        }
-        if(tensor_dim == 5)
+
+        switch(miopen::deref(tensorDesc).GetType())
```
Comment (junliume) [inline src/pooling_api.cpp:45]: [2022-07-26T19:02:47.992Z] /home/jenkins/workspace/MIOpen_ddu_fix_pooling_cmd_print/src/pooling_api.cpp:45:16: error: 5 enumeration values not explicitly handled in switch: 'miopenInt32', 'miopenInt8', 'miopenInt8x4'... [clang-diagnostic-switch-enum,-warnings-as-errors]

[2022-07-26T19:02:47.992Z]         switch(miopen::deref(tensorDesc).GetType())

[2022-07-26T19:02:47.992Z]                ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

[2022-07-26T19:02:47.992Z] src/CMakeFiles/tidy-target-MIOpen-pooling_api_cpp.dir/build.make:57: recipe for target 'src/CMakeFiles/tidy-target-MIOpen-pooling_api_cpp' failed

**Comment by DrizztDoUrden** [comment]
@jun that line won't be parced by the library at all since the command line args update. The correct line would be like this:
```
/root/MIOpen/build# ./bin/MIOpenDriver pool -M 0 --input 2x2x1x5 -y 1 -x 1 -p 0 -q 0 -v 1 -u 1 -m avg -F 1 -t 1
```
My fix doesn't change the arguments, it changes the printout to the correct format. You could use short key instead of input, but i don't remember which one, should be in --help if you care, but that makes it unreadable for anyone unfamiliar with the syntax.

File: src/pooling_api.cpp [hunk 2]
```diff
@@ -46,6 +46,11 @@ inline void Pooling_logging_cmd(const miopenPoolingDescriptor_t poolDesc,
         {
         case miopenHalf: ss << "poolfp16"; break;
         case miopenFloat: ss << "pool"; break;
+        case miopenInt32:
+        case miopenInt8:
+        case miopenInt8x4:
+        case miopenBFloat16:
+        case miopenDouble:
         default: MIOPEN_THROW(miopenStatusNotImplemented);
```
Comment (atamazov) [inline src/pooling_api.cpp:54]: Let's avoid fatal errors in the debugging code. It is the library core that should do that. Here I recommend printing things like "poolint8_UNIMPLEMENTED", "poolbfp16_UNIMPLEMENTED", "poolfp64_UNIMPLEMENTED".

**Review by atamazov — CHANGES_REQUESTED** [review]

**Comment by atamazov** [comment]
@junliume 
> ```
> MIOpenDriver pool -M 0 -n 2 -c 2 -H 1 -W 5 -y 1 -x 1 -p 0 -q 0 -v 1 -u 1 -m avg -F 1 -t 1
> Input Flag: n Not Found !
> ```
> 
> anything I might have missed here?

Please look at #1313 for more info about the new syntax.

**Comment by junliume** [comment]
> @junliume
> 
> > ```
> > MIOpenDriver pool -M 0 -n 2 -c 2 -H 1 -W 5 -y 1 -x 1 -p 0 -q 0 -v 1 -u 1 -m avg -F 1 -t 1
> > Input Flag: n Not Found !
> > ```  
> >   
> > anything I might have missed here?
> 
> Please look at #1313 for more info about the new syntax.

Thanks, @atamazov !

These lines are grepped from the debug output from https://github.com/ROCmSoftwarePlatform/MIOpen/issues/1642

So I guess something need to be changed in the debug output?

BTW~ #1642 is still an urgent issue :(

**Comment by atamazov** [comment]
@junliume We are working on #1642 together with @DrizztDoUrden

File: src/pooling_api.cpp [hunk 3]
```diff
@@ -85,15 +97,17 @@ inline void Pooling_logging_cmd(const miopenPoolingDescriptor_t poolDesc,
...
+           << (tensor_dim == 5 ? miopen::deref(poolDesc).strides[1]
+                               : miopen::deref(poolDesc).strides[0])
+           << " -u "
+           << (tensor_dim == 5 ? miopen::deref(poolDesc).strides[2]
+                               : miopen::deref(poolDesc).strides[1])
+           << " -m "
+           << (miopen::deref(poolDesc).mode == 0
+                   ? "max"
+                   : (miopen::deref(poolDesc).mode == 1 ? "avg" : "avg_in"))
+           << " -F " << ((is_fwd) ? "1" : "2") << " -t 1"; // clang-format on
```
Comment (atamazov) [inline src/pooling_api.cpp:118]: If `-F 2` is not actually functional, shall we better output `-F 0` instead?

File: src/pooling_api.cpp (hunk 2 above)
Comment (atamazov) [reply src/pooling_api.cpp]: Resolved.

**Comment by DrizztDoUrden** [comment]
@junliume As it only changes log lines not even used in the tests, I don't think we need to spend CI time for the rest of the tests, but the PR is no hurry either.

**Comment by DrizztDoUrden** [comment]
@junliume this has passed the CI

**Review by junliume — APPROVED** [review]

**Pull request merged**